openlibrary/macros/databarWork.html

```diff
@@ -123,7 +123,7 @@ <h3 class="closed">
     </div>
     </div>
 
-$if isbn or oclc_numbers or 'lendinglibrary' in collection or library:
+$if (isbn or oclc_numbers or 'lendinglibrary' in collection or library) and not any_private_collections(collection):
```

Comment:
It doesn't look like it is smart enough to optimize it away; e.g.:

    >>> import timeit
    >>> timeit.timeit('do_something_expensive = lambda: range(100);[do_something_expensive() for x in range(10) if do_something_expensive()]')
    26.7s

    >>> timeit.timeit('do_something_expensive = lambda: range(100);[do_something_expensive() for x in range(10) if True]')
    13.81285309791565